dao/src/main/java/org/thingsboard/server/dao/sql/query/EntityKeyMapping.java

```diff
@@ -537,6 +537,7 @@ private String buildStringPredicateQuery(QueryContext ctx, String field, StringF
             value = value.toLowerCase();
             operationField = String.format("lower(%s)", operationField);
         }
+        List<String> values = new ArrayList<>();
```

Comment:
let's remove this array that is not related to all cases

Reply:
we can create this array in second switch statement

Reply:
case IN:
            case NOT_IN:
                ctx.addStringListParameter(paramName, List.of(value.trim().split("\\s*,\\s*")));
                break;